src/instructlab/rag/process_docs.py

```diff
@@ -0,0 +1,156 @@
+# Standard
+from pathlib import Path
+from typing import Iterable
+import json
+import logging
+import os
+import tempfile
+import time
+
+# Third Party
+from docling.datamodel.base_models import ConversionStatus  # type: ignore
+from docling.datamodel.base_models import InputFormat  # type: ignore
+from docling.datamodel.document import ConversionResult  # type: ignore
+from docling.datamodel.pipeline_options import EasyOcrOptions  # type: ignore
+from docling.datamodel.pipeline_options import OcrOptions  # type: ignore
+from docling.datamodel.pipeline_options import PdfPipelineOptions  # type: ignore
+from docling.datamodel.pipeline_options import TesseractOcrOptions  # type: ignore
+from docling.document_converter import DocumentConverter  # type: ignore
+from docling.document_converter import PdfFormatOption  # type: ignore
+from docling.models.easyocr_model import EasyOcrModel  # type: ignore
+from docling.models.tesseract_ocr_model import TesseractOcrModel  # type: ignore
+
+# First Party
+from instructlab.rag.taxonomy_utils import lookup_knowledge_files
+from instructlab.utils import clear_directory
+
+logger = logging.getLogger(__name__)
+
+
+def process_docs_from_taxonomy(taxonomy_path, taxonomy_base, output_dir):
+    # Taxonomy navigation strategy:
+    # Create temp folder that is deleted when the function returns
+    # Read taxonomy using read_taxonomy_leaf_nodes from instructlab-sdg package
+    # The above step also downloads the reference documents.
+    # Move all the downloaded documents under the temp folder
+    # Pass the list to process_docs_from_folder
+    with tempfile.TemporaryDirectory() as temp_dir:
+        logger.info(f"Temporary directory created: {temp_dir}")
+        knowledge_files = lookup_knowledge_files(taxonomy_path, taxonomy_base, temp_dir)
+        logger.info(f"Found {len(knowledge_files)} knowledge files")
+        logger.info(f"{knowledge_files}")
+
+        process_docs_from_folder(temp_dir, output_dir)
+
+
+# Copied from sdg.utils.chunkers because that code is being refactored so we want to avoid importing anything from it.
+# TODO: Once the code base has settled down, we should make sure this code exists only in one place.
+def resolve_ocr_options() -> OcrOptions:
```

Comment:
seems this is a "private" function, maybe we can rename to `_resolve_ocr_options`?